ui/views.py

```diff
@@ -117,13 +118,21 @@ def edit_vocabulary(request, repo_slug, vocab_slug):
 
 
 @login_required
+@permission_required_or_403(
```

Comment:
Okay, just an idle thought since if they're not logged in they can't have any permissions. Don't worry about it. If it'll return different status codes it's probably not worth it.